test/dtypes/test_uint2.py

```diff
@@ -0,0 +1,47 @@
+from unittest import main
+
+import torch
+import torch.nn as nn
+
+from torch.testing._internal.common_quantization import (
+    NodeSpec as ns,
+    QuantizationTestCase,
+)
+
+
+from torchao.dtypes.uint2 import (
+    UInt2Tensor,
+    BitnetTensor
+)
+from torchao.quantization.quant_api import (
+    _replace_with_custom_fn_if_matches_filter,
+)
+
+def _apply_weight_only_uint2_quant(model):
+    def fn(mod):
+        mod.weight = torch.nn.Parameter(BitnetTensor.from_float(mod.weight), requires_grad=False)
+        return mod
+
+    _replace_with_custom_fn_if_matches_filter(
+        model,
+        lambda mod: fn(mod),
+        lambda mod, fqn: isinstance(mod, torch.nn.Linear),
+    )
+
+
+class TestUInt2(QuantizationTestCase):
+    def test_gpu_quant(self):
+        for x_shape in [[2, 4], [5, 5, 5, 4], [1, 4, 4]]:
+            x = torch.randn(*x_shape)
+            m = nn.Sequential(nn.Linear(4, 16))
+            y_ref = m(x)
+            _apply_weight_only_uint2_quant(m)
+            y_wo = m(x)
+            # sqnr = compute_error(y_ref, y_wo)
+            #opt = torch.compile(m, fullgraph=True, mode="max-autotune")
```

Comment:
What's the error you were getting?

Reply:
AssertionError: Please convert all Tensors to FakeTensors first or instantiate FakeTensorMode with 'allow_non_fake_inputs'. Found in aten.sym_storage_off.set.default(_to_functional_tensor(FakeTensor(..., size=(16, 4), dtype=torch.uint8)))
It generally failed enabling @torch.compile on pack and unpack functions

Reply:
@bdhirsh any ideas?

Reply:
opened this issue https://github.com/pytorch/pytorch/issues/127374